angr/code_location.py

```diff
@@ -80,11 +80,14 @@ def __eq__(self, other):
         return type(self) is type(other) and self.block_addr == other.block_addr and \
                 self.stmt_idx == other.stmt_idx and self.sim_procedure is other.sim_procedure and \
                 self.context == other.context and \
-                self.block_idx == other.block_idx
+                self.block_idx == other.block_idx and \
+                self.ins_addr
 
     def __lt__(self, other):
         if self.block_addr != other.block_addr:
             return self.block_addr < other.block_addr
+        if self.ins_addr != other.ins_addr:
+            return self.ins_addr < other.ins_addr
```

Comment:
I think we should make it so that we only compare `self.ins_addr` against `other.ins_addr` if they are both not `None`.